packages/terra-doc-template/src/DocTemplate.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import classNames from 'classnames/bind';
+import IndexExampleTemplate from './IndexPageTemplate';
+import styles from './DocTemplate.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The given component's npm package name
+   */
+  packageName: PropTypes.string,
+  /**
+   * The given component's readme file imported to a string
+   */
+  readme: PropTypes.string,
+  /**
+   * All of the example(s) that will be displayed. An empty array is supported.
+   * ```
+   * title: The title of the example
+   * description: A description of the example to be displayed below the title
+   * example: The example to be displayed
+   * source: The source code of the example
+   * ```
+   */
+  examples: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        title: PropTypes.string,
+        description: PropTypes.string,
+        example: PropTypes.element,
+        source: PropTypes.string,
+      },
+    ),
+  ),
+  /**
+   * All of the props table(s) that will be displayed. An empty array is supported.
+   * ```
+   * componentSrc: The source code of the component
+   * componentName: The name of the component
+   * ```
+   */
+  propsTables: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        componentSrc: PropTypes.string,
+        componentName: PropTypes.string,
+      },
+    ),
+  ),
+};
+
+const defaultProps = {
+  packageName: '',
+  readme: '',
+  examples: [],
+  propsTables: [],
+};
+
+const DocTemplate = ({ packageName, readme, examples, propsTables, ...customProps }) => {
+  let id = 0;
+  const localExamples = examples;
+  const localPropsTables = propsTables;
+
+  // Assign unique identifiers to use as keys later
+  for (let i = 0; i < localExamples.length; i += 1) {
+    localExamples[i].id = id;
+    id += 1;
+  }
+
+  for (let i = 0; i < localPropsTables.length; i += 1) {
+    localPropsTables[i].id = id;
+    id += 1;
+  }
+
+  const version = `[![NPM version](http://img.shields.io/npm/v/${packageName}.svg)](https://www.npmjs.org/package/${packageName})`;
+
+  return (
+    <div {...customProps}>
+      {packageName && <Markdown src={version} />}
+      {readme && <Markdown src={readme} />}
+
+      {localExamples.length > 0 && <h1 className={cx('.examples-header')} >Examples</h1>}
+      {localExamples.map(example =>
```

Comment:
When using the [map function](https://www.w3schools.com/jsref/jsref_map.asp), it is possible to access the current index of the value being 'mapped'. This can be updated to
```
{localExamples.map((example, index) =>
    <IndexExampleTemplate
       title={example.title}
       example={example.example}
       exampleSrc={example.source}
       description={example.description}
       key={index}
    />,
)}
``` 
This would remove the need to map the index to example.id above.

Reply:
Eslint doesn't like that @emilyrohrbough : `[eslint] Do not use Array index in keys (react/no-array-index-key)`

Reply:
The linter doesn't like any inclusion of the index at all, even when combined with title. We could use title as the key but they aren't required to be unique, or required at all.